spring-integration-gcp/src/main/java/org/springframework/integration/gcp/pubsub/outbound/PubSubMessageHandler.java

```diff
@@ -96,6 +99,10 @@ else if (payload instanceof ByteString) {
 		ListenableFuture<String> pubsubFuture =
 				this.pubSubTemplate.publish(this.topic, pubsubPayload, headers);
 
+		if (this.publishFutureCallback != null) {
+			pubsubFuture.addCallback(this.publishFutureCallback);
```

Comment:
We may want to just add a logging one when none is specified. In that case we wouldn't want the logging done in the template. I'm not sure if that is what @artembilan was thinking of when he mentioned `LoggingListenableFutureCallback`.

Reply:
Yeah... If we have logging on the `template` level, we don't need it on the adapter.
Since `template` can even be used outside of the SI, I don't see reason to overhead with extra default callback.
Current state is fine with me.

Reply:
sg